lib/git-user.js

```diff
@@ -9,7 +9,7 @@ module.exports = function () {
     email = exec('git config --get user.email')
   } catch (e) {}
 
-  name = name && name.toString().trim()
+  name = name && JSON.stringify(name.toString().trim()).slice(1, -1)
```

Comment:
@sayanriju `JSON.stringify('hello')` will result in an additional set of quotes around the output:

```javascript
console.log(JSON.stringify('hey'))
// "hey"
```

i.e.: the resultant string is actually `"hey"` as opposed to just `hey`

In our case, we don't want the resultant Git `user.name` to be wrapped in quotes but we would like to escape them:

```javascript
var gitConfigName = 'hello "me" you'
var packageJSON = { author: JSON.stringify('hello "me" you').slice(1, -1) }
console.log(packageJSON)
// { author: 'hello \\"me\\" you' }
packageJSON = { author: JSON.stringify('hello "me" you') }
console.log(packageJSON)
// { author: '"hello \\"me\\" you"' }
```

I hope that helps clear it up a bit.